HTTPServer.driver/PutCriteria.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using HTTPServer.core;
+
+namespace HTTPServer.app
+{
+    public class PutCriteria : ICriteria
+    {
+        public bool ShouldRun(Request request)
+        {
+            return request.Method.Equals("PUT") && request.HttpVersion.Equals("HTTP/1.1") && !request.Uri.Equals("/");
```

Comment:
I could take the version check out of all of these and the code would run the same.  The reason is I check the errors first.  The reason that it is in all of these is that all of them is it makes each Command independent of the other Commands.  Could I make the assumption that errors are always going to be checked first?  If not, the place I could check for it is before it goes into the application.  Should the server know which version is being used?  It seems like the server should be ignorant of those details.